lib/lumberjack/server.rb

```diff
@@ -42,31 +42,42 @@ def initialize(options={})
       @ssl.key = OpenSSL::PKey::RSA.new(File.read(@options[:ssl_key]),
                                         @options[:ssl_key_passphrase])
       @ssl_server = OpenSSL::SSL::SSLServer.new(@tcp_server, @ssl)
+      @stop = false
     end # def initialize
 
     def run(&block)
-      while true
-        connection = accept
-        Thread.new(connection) do |connection|
+      while !@stop
+        sock = @tcp_server.accept
+        Thread.new(sock) do |sock|
+          connection = accept_ssl(sock)
```

Comment:
Thinking about this more, I think this should use a threadpool instead of doing the handshake in the connection thread.

Using a pool, instead of per-connection-thread, to accept+handshake can help prevent resource starvation of active connections by allowing the pool size to be fixed-size and configurable.